worldsock.cpp

```
@@ -28,32 +28,72 @@ IMPLEMENT_DYNAMIC(CWorldSocket, CAsyncSocket)
+      // if the socket was destroyed or replaced, don't touch the old object
+      if (pDoc->m_pSocket != this)
         return;
+
+      // SSL may have buffered more decrypted data than one SSL_read consumed.
+      // Since we won't get another FD_READ for already-buffered data, drain it now.
+      if (pDoc->m_pSSL && pDoc->m_bSSL_Connected)
+        {
+        while (SSL_pending (pDoc->m_pSSL) > 0)
+          {
+          pDoc->ReceiveMsg();
+          if (pDoc->m_pSocket != this)
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)